docs/src/main/asciidoc/resteasy-reactive.adoc

```diff
@@ -1150,6 +1150,19 @@ class Filters {
 }
 ----
 
+[IMPORTANT]
+====
+Request filters are usually executed on the same thread that the method that handles the request will be executed.
+That means that if the method servicing the request is annotated with `@Blocking`, then the filters will also be run
+on the worker thread.
+If the method is annotated with `@NonBlocking` (or is not annotated at all), then the filters will also be run
+on the same event-loop thread.
+
+If however a filter needs to be run on the event-loop despite the fact that the method servicing the request will be
+run on a worker thread, then `@ServerRequestFilter(nonBlocking=true)` can be used.
```

Comment:
So we don't support `@NonBlocking` on JAXRS-style filters? BTW, why not use `@NonBlocking` here instead of an annotation parameter?

Reply:
I chose to use an annotation parameter to keep it consistent with what we do for `priority`

Reply:
OK, so this begs the question of why we do it for priority? :)

Reply:
We do support the annotation for the normal JAX-RS filters

Reply:
`Priority` has `@Target({TYPE,PARAMETER})` that's why.

But this annotation doesn't…
![choice](https://user-images.githubusercontent.com/179265/111822788-8037e580-88e4-11eb-91fc-b00ff1c1bd87.jpeg)

Reply:
Hahaha!

I would favor consistency

Reply:
> We do support the annotation for the normal JAX-RS filters

OK good. Probably don't need to mention it here as we discourage people from using them.

Reply:
Yeah, exactly